erts/emulator/test/persistent_term_SUITE.erl

```diff
@@ -960,12 +984,18 @@ put_erase_trapping(_Config) ->
     do_erases(NrOfItems),
     ok.
 
-do_puts(0, _) -> ok;
 do_puts(NrOfPuts, ValuePrefix) ->
+    do_puts_iter(NrOfPuts, ValuePrefix, put).
+
+do_put_news(NrOfPuts, ValuePrefix) ->
+    do_puts_iter(NrOfPuts, ValuePrefix, put_new).
+
+do_puts_iter(0, _, _) -> ok;
+do_puts_iter(NrOfPuts, ValuePrefix, PutFun) ->
     Key = {?MODULE, NrOfPuts},
     Value = {ValuePrefix, NrOfPuts},
-    erts_debug:set_internal_state(reds_left, rand:uniform(250)),
-    persistent_term:put(Key, Value),
+    %% erts_debug:set_internal_state(reds_left, rand:uniform(250)),
```

Comment:
For some reason the trapping tests fail due to a badarg from `get/1` when this is uncommented. This badarg seems to magically disappear when this line is commented.

I don't understand why changing the number of reductions left would cause `put/2` and `put_new/2` to effectively not execute. Does this point towards some error in my trapping?

Also I don't understand why this error only showed up in my last change and not in any earlier ones.

Reply:
It's these lines:

```erlang
erts_debug:set_internal_state(reds_left, rand:uniform(250)),
erlang:apply(persistent_term, PutFun, [Key, Value]),
Value = persistent_term:get(Key),
```

`get/1` fails with the `badarg`, so I'm assuming that's because `put/2` and `put_new/2` don't actually update.